tests/test_bootstrap.py

```diff
@@ -0,0 +1,1390 @@
+            )
+            obj = ObjectiveFunction(
+                BootstrapRedlConsistency(
+                    grid=grid,
+                    helicity=helicity,
+                    rho_exponent=1,
+                ),
+                eq,
+            )
+            scalar_objective = obj.compute_scalar(obj.x(eq))
+            return scalar_objective
+
+        results = []
+
+        # Loop over grid trypes. For LinearGrid we need to drop the
+        # point at rho=0 to avoid a divide-by-0
+        grid_types = [LinearGrid, QuadratureGrid]
+        kwargss = [{"axis": False}, {}]
+
+        # Loop over grid resolutions:
+        Ls = [10, 20, 10, 10]
+        Ms = [10, 10, 20, 10]
+        Ns = [0, 0, 0, 2]
+
+        for grid_type, kwargs in zip(grid_types, kwargss):
+            for L, M, N in zip(Ls, Ms, Ns):
+                results.append(test(grid_type, kwargs, L, M, N))
+
+        results = np.array(results)
+        np.testing.assert_allclose(results, np.mean(results), rtol=0.02)
+
+    # MJL: Should this test be decorated with @pytest.mark.slow ?
+    @pytest.mark.unit
+    def test_bootstrap_consistency_iota(self):
+        """Try optimizing for bootstrap consistency in axisymmetry, at fixed shape.
+
+        This version of the test covers an equilibrium with an iota
+        profile rather than a current profile.
+        """
+        ne0 = 4.0e20
+        T0 = 12.0e3
+        ne = PowerSeriesProfile(ne0 * np.array([1, -1]), modes=[0, 10])
+        Te = PowerSeriesProfile(T0 * np.array([1, -1]), modes=[0, 2])
+        Ti = Te
+        Zeff = 1
+
+        helicity = (1, 0)
+        B0 = 5.0  # Desired average |B|
+        LM_resolution = 8
+
+        initial_iota = -0.61
+        num_iota_points = 21
+        iota = SplineProfile(np.full(num_iota_points, initial_iota))
+
+        # Set initial condition:
+        Rmajor = 6.0
+        aminor = 2.0
+        NFP = 1
+        surface = FourierRZToroidalSurface(
+            R_lmn=np.array([Rmajor, aminor]),
+            modes_R=[[0, 0], [1, 0]],
+            Z_lmn=np.array([-aminor]),
+            modes_Z=[[-1, 0]],
+            NFP=NFP,
+        )
+
+        eq = Equilibrium(
+            surface=surface,
+            electron_density=ne,
+            electron_temperature=Te,
+            ion_temperature=Ti,
+            atomic_number=Zeff,
+            iota=iota,
+            Psi=B0 * np.pi * (aminor**2),
+            NFP=NFP,
+            L=LM_resolution,
+            M=LM_resolution,
+            N=0,
+            L_grid=2 * LM_resolution,
+            M_grid=2 * LM_resolution,
+            N_grid=0,
+            sym=True,
+        )
+
+        eq.solve(
+            verbose=3,
+            ftol=1e-8,
+            constraints=get_fixed_boundary_constraints(kinetic=True),
+            optimizer=Optimizer("lsq-exact"),
+            objective=ObjectiveFunction(objectives=ForceBalance()),
+        )
+
+        eq.save("test_bootstrap_consistency_iota_initial.h5")
```

Comment:
Should preferably save to a temporary directory if needed (using `TmpDir` fixture) so we don't generate a bunch of local files when running tests

Reply:
Now `TmpDir` is used.